app/src/app/api/v0/city/[city]/inventory/[year]/route.ts

```diff
@@ -18,9 +20,76 @@ export const GET = apiHandler(async (_req: NextRequest, { params }) => {
   }
   inventory.city = city;
 
-  return NextResponse.json({ data: inventory });
+  let body:Buffer|null = null;
+  let headers:Record<string, string>|null = null;
+
+  switch (_req.nextUrl.searchParams.get('format')?.toLowerCase()) {
+    case "csv":
+      body = await inventoryCSV(inventory);
+      headers = {
+        "Content-Type": "text/csv",
+        "Content-Disposition": `attachment; filename="inventory-${inventory.city.locode}-${inventory.year}.csv"`,
+      }
+      break;
+    case "xls":
+      body = await inventoryXLS(inventory);
+      headers = {
+        "Content-Type": "application/vnd.ms-excel",
+        "Content-Disposition": `attachment; filename="inventory-${inventory.city.locode}-${inventory.year}.xls"`,
+      }
+      break;
+    case "json":
+    default:
+      body = Buffer.from(JSON.stringify({data: inventory.toJSON()}), "utf-8");
+      headers = {
+        "Content-Type": "application/json"
+      }
+      break;
+  }
+
+  return new NextResponse(body, {headers});
 });
 
+async function inventoryCSV(inventory: any) : Promise<Buffer> {
+  const subSectorValues:SubSectorValue[] =
+    await inventory.getSubSectorValues();
+  const subCategoryValues:SubCategoryValue[] =
+    await inventory.getSubCategoryValues();
+  const headers = [
+    "Inventory Reference",
+    "Total Emissions",
+    "Activity Units",
+    "Activity Value",
+    "Emission Factor Value",
+    "Datasource ID"
+  ].join(",");
+  const subSectorLines = subSectorValues.map((value:SubSectorValue) => {
+    return [
+      value.subsectorId,
+      value.totalEmissions,
+      value.activityUnits,
+      value.activityValue,
+      value.emissionFactorValue,
+      value.datasourceId
+    ].join(",");
+  });
+  const subCategoryLines = subCategoryValues.map((value:SubCategoryValue) => {
+    return [
+      value.subcategoryId,
+      value.totalEmissions,
+      value.activityUnits,
+      value.activityValue,
+      value.emissionFactorValue,
+      value.datasourceId
+    ].join(",");
+  });
+  return Buffer.from([headers, ...subSectorLines, ...subCategoryLines].join("\n"));
```

Comment:
Should this maybe be structured as 1 subsector line, then all of its subcategories, then the next subsector? We will add the GPC reference number (see [this PR](https://github.com/Open-Earth-Foundation/CityCatalyst/pull/145)) to the SubSector/ SubCategory data model, so it could be used instead of the values for the first column here, which will just be UUIDs that won't mean much to an end user (unless it needs to be imported into CC again for some reason...)

Reply:
Yeah, I think the sorting there can be handled by the downloader.